pkg/controllers/clusterresourceplacementeviction/controller.go

```diff
@@ -277,13 +283,9 @@ func isEvictionAllowed(bindings []placementv1beta1.ClusterResourceBinding, crp p
 
 	var desiredBindings int
 	placementType := crp.Spec.Policy.PlacementType
-	switch placementType {
-	case placementv1beta1.PickNPlacementType:
+	// we don't know the desired bindings for PickAll and we won't evict a binding for PickFixed CRP.
+	if placementType == placementv1beta1.PickNPlacementType {
 		desiredBindings = int(*crp.Spec.Policy.NumberOfClusters)
-	case placementv1beta1.PickFixedPlacementType:
-		desiredBindings = len(crp.Spec.Policy.ClusterNames)
-	default:
-		// we don't know the desired bindings for PickAll.
 	}
 
 	var disruptionsAllowed int
```

Comment:
I can't seem to find where  "For PickAll CRPs, MaxUnavailable won't be specified in DB" is enforced?

Reply:
https://github.com/Azure/fleet/blob/main/pkg/controllers/clusterresourceplacementeviction/controller.go#L195-L201

Reply:
so we rejected any eviction associated with the PDB instead of rejecting the PBD itself

Reply:
We use PDB as an information source as of now, even if the CRP targeted by PDB is missing we allow the PDB to exist. So in the case of,

1. PickAll CRP where the PDB specifies MaxUnavailable or MinAvailable as percentage we say the eviction cannot be executed due to a misconfigured PDB
2. In the case for PickFixed CRP, the eviction fails on validation, and we mark eviction as invalid